### ISSUE TITLE
Wrong index calculation in constructing sparse random matrix 

### ISSUE THREAD

**Issue by WiseroOrb** [cupy/cupy#6255]
### Description

In cupyx/scipy/sparse/construct.py the constructor random first generates the indices as a contiguous array, than transforms them to row and column indices. However to do that uses a formula that does not work for certain values:

line: 367
```py
ind = random_state.choice(mn, size=k, replace=False)
j = cupy.floor(ind * (1. / m)).astype('i')
i = ind - j * m
```

For example if ind = 3150 and m = 3150, j should be 1, but this algorithm returns 0, probably for rounding errors. In this case i = 3150, and this causes a 
ValueError('row index exceeds matrix dimensions') later in the coo constructor

The simple fix is to calculate directly the integer division, with the notation //, so the code becomes:

```py
ind = random_state.choice(mn, size=k, replace=False)
j = ind//m
i = ind - j * m
```

 

### Installation

Source (`pip install cupy`)

### Environment

```
OS                           : Linux-5.13.0-22-generic-x86_64-with-glibc2.34
Python Version               : 3.9.7
CuPy Version                 : 10.0.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 1.20.3
SciPy Version                : 1.7.1
Cython Build Version         : 0.29.24
Cython Runtime Version       : None
CUDA Root                    : /usr/local/cuda
nvcc PATH                    : /usr/local/cuda/bin/nvcc
CUDA Build Version           : 11040
CUDA Driver Version          : 11040
CUDA Runtime Version         : 11040
cuBLAS Version               : (available)
cuFFT Version                : 10501
cuRAND Version               : 10205
cuSOLVER Version             : (11, 2, 0)
cuSPARSE Version             : (available)
NVRTC Version                : (11, 4)
Thrust Version               : 101201
CUB Build Version            : 101201
Jitify Build Version         : 60e9e72
cuDNN Build Version          : (not loaded; try `import cupy.cuda.cudnn` first)
cuDNN Version                : (not loaded; try `import cupy.cuda.cudnn` first)
NCCL Build Version           : None
NCCL Runtime Version         : None
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : NVIDIA GeForce GTX 1050 Ti with Max-Q Design
Device 0 Compute Capability  : 61
Device 0 PCI Bus ID          : 0000:01:00.0

```

**Comment by emcastillo** [comment]
Thank you! Can you send a PR?

**Comment by gajendra0180** [comment]
Hey! I would like to fix this bug if nobody has yet started working on it. 
Thanks.

**Comment by WiseroOrb** [comment]
Go ahead

**Comment by gajendra0180** [comment]
i am not able to set it up on my local system.

> [Requirements
> NVIDIA CUDA GPU with the Compute Capability 3.0 or larger.]

Can it be installed without  NVIDIA CUDA GPU?


Log:
```

-------- Configuring Module: cuda --------
Command "x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -D_FORCE_INLINES=1 -I/home/error0180/git_projects/cupy/cupy/_core/include/cupy/cub -I/home/error0180/git_projects/cupy/cupy/_core/include -c /tmp/tmpbimo8oej/a.cpp -o /tmp/tmpbimo8oej/tmp/tmpbimo8oej/a.o" failed with exit status 1
**************************************************
*** WARNING: Cannot check compute capability
Command "x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/home/error0180/git_projects/cupy/cupy/_core/include/cupy/cub -I/home/error0180/git_projects/cupy/cupy/_core/include -c /tmp/tmpgsfyt7in/a.cpp -o /tmp/tmpgsfyt7in/tmp/tmpgsfyt7in/a.o" failed with exit status 1
**************************************************

************************************************************
* CuPy Configuration Summary                               *
************************************************************

Build Environment:
  Include directories: ['/home/error0180/git_projects/cupy/cupy/_core/include/cupy/cub', '/home/error0180/git_projects/cupy/cupy/_core/include']
  Library directories: []
  nvcc command       : (not found)
  hipcc command      : (not found)

Environment Variables:
  CFLAGS          : (none)
  LDFLAGS         : (none)
  LIBRARY_PATH    : (none)
  CUDA_PATH       : (none)
  NVTOOLSEXT_PATH : (none)
  NVCC            : (none)
  HIPCC           : (none)
  ROCM_HOME       : (none)

Modules:
  cuda      : No
    -> Include files not found: ['cublas_v2.h', 'cuda.h', 'cuda_profiler_api.h', 'cuda_runtime.h', 'cufft.h', 'curand.h', 'cusparse.h', 'nvrtc.h']
    -> Check your CFLAGS environment variable.

ERROR: CUDA could not be found on your system.

HINT: You are trying to build CuPy from source, which is NOT recommended for general use.
      Please consider using binary packages instead.

Please refer to the Installation Guide for details:
https://docs.cupy.dev/en/stable/install.html

************************************************************

Traceback (most recent call last):
  File "setup.py", line 85, in <module>
    ext_modules = cupy_setup_build.get_ext_modules(False, ctx)
  File "/home/error0180/git_projects/cupy/install/cupy_builder/cupy_setup_build.py", line 515, in get_ext_modules
    extensions = make_extensions(ctx, compiler, use_cython)
  File "/home/error0180/git_projects/cupy/install/cupy_builder/cupy_setup_build.py", line 319, in make_extensions
    raise Exception('Your CUDA environment is invalid. '
Exception: Your CUDA environment is invalid. Please check above error log.

```

I installed _cupy-cuda101,_ and _cupy-cuda113,_ but still the same problem persist.

I searched on web and came through some of these links but still same problem persist:

https://github.com/explosion/spaCy/issues/5368
https://stackoverflow.com/questions/50009625/cupy-error-installation
https://askubuntu.com/questions/210884/setting-ld-library-path-for-cuda

Thanks.

**Comment by WiseroOrb** [comment]
#6257

**Comment by WiseroOrb** [comment]
PR 
#6257

**Comment by gajendra0180** [comment]
Thanks.

**Comment by kmaehashi** [comment]
Fixed via #6257. Thanks!
